pkg/agent/monitortool/latency_store_test.go

```diff
@@ -0,0 +1,227 @@
+			assert.Equal(t, tt.expectedEntry.LastSendTime, entry.LastSendTime)
+			assert.Equal(t, tt.expectedEntry.LastRecvTime, entry.LastRecvTime)
+		})
+	}
+}
+
+func TestLatencyStore_DeleteNodeIPLatencyEntry(t *testing.T) {
+	tests := []struct {
+		key               string
+		prevExpectedEntry NodeIPLatencyEntry
+		expectedEntry     NodeIPLatencyEntry
+	}{
+		{
+			key:               "10.244.2.1",
+			prevExpectedEntry: entry,
+			expectedEntry:     NodeIPLatencyEntry{},
+		},
+		{
+			key:               "10.244.2.2",
+			prevExpectedEntry: NodeIPLatencyEntry{},
+			expectedEntry:     NodeIPLatencyEntry{},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.key, func(t *testing.T) {
+			latencyStore := &LatencyStore{
+				isNetworkPolicyOnly: false,
+				nodeIPLatencyMap: map[string]*NodeIPLatencyEntry{
+					"10.244.2.1": &entry,
+				},
+				nodeTargetIPsMap: map[string][]net.IP{
+					"Node1": {net.ParseIP("10.244.2.1")},
+				},
+			}
+
+			entry, _ := latencyStore.getNodeIPLatencyEntry(tt.key)
+			assert.Equal(t, tt.prevExpectedEntry, entry)
+			latencyStore.DeleteNodeIPLatencyEntry(tt.key)
+			entry, ok := latencyStore.getNodeIPLatencyEntry(tt.key)
+			assert.Equal(t, entry, tt.expectedEntry)
+			assert.False(t, ok)
+		})
+	}
+}
+
+func TestLatencyStore_SetNodeIPLatencyEntry(t *testing.T) {
+	tests := []struct {
+		key           string
+		updatedEntry  NodeIPLatencyEntry
+		expectedEntry NodeIPLatencyEntry
+	}{
+		{
+			key:           "10.244.2.1",
+			updatedEntry:  entry,
+			expectedEntry: entry,
+		},
+		{
+			key:           "10.244.2.1",
+			updatedEntry:  entry2,
+			expectedEntry: entry2,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.key, func(t *testing.T) {
+			latencyStore := &LatencyStore{
+				isNetworkPolicyOnly: false,
+				nodeIPLatencyMap: map[string]*NodeIPLatencyEntry{
+					"10.244.2.1": &entry,
+				},
+				nodeTargetIPsMap: map[string][]net.IP{
+					"Node1": {net.ParseIP("10.244.2.1")},
+				},
+			}
+
+			mutator := func(entry *NodeIPLatencyEntry) {
+				entry.LastSendTime = tt.updatedEntry.LastSendTime
+				entry.LastRecvTime = tt.updatedEntry.LastRecvTime
+				entry.LastMeasuredRTT = tt.updatedEntry.LastMeasuredRTT
+			}
+			latencyStore.SetNodeIPLatencyEntry(tt.key, mutator)
+			entry, ok := latencyStore.getNodeIPLatencyEntry(tt.key)
+			assert.Equal(t, tt.expectedEntry, entry)
+			assert.True(t, ok)
+		})
+	}
+}
+
+func TestLatencyStore_DeleteStaleNodeIPs(t *testing.T) {
+	testKey := "10.244.2.1"
+
+	latencyStore := &LatencyStore{
+		isNetworkPolicyOnly: false,
+		nodeIPLatencyMap: map[string]*NodeIPLatencyEntry{
+			"10.244.2.1": &entry,
+		},
+		nodeTargetIPsMap: map[string][]net.IP{
+			"Node1": {net.ParseIP("10.244.2.1")},
+		},
+	}
+
+	// Remove node
+	latencyStore.deleteNode(&corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "Node1",
+		},
+	})
+
+	// Check if the entry is deleted
+	_, ok := latencyStore.getNodeIPLatencyEntry(testKey)
+	assert.False(t, ok)
```

Comment:
This should be `True` or the test is wrong. If this is `False`, then `DeleteStaleNodeIPs` is essentially not being used, as the entry has already been deleted.

Reply:
Ok, it really should be True here, in the old implementation deleteNode was used to delete the NodeIPLatencyEntry, now the `deleteNode` function has been modified.

Reply:
```bash
antrea/pkg/agent/monitortool$ go test ./... -v -race
=== RUN   TestLatencyStore_getNodeIPLatencyEntry
=== RUN   TestLatencyStore_getNodeIPLatencyEntry/10.244.2.1
=== RUN   TestLatencyStore_getNodeIPLatencyEntry/10.244.2.2
--- PASS: TestLatencyStore_getNodeIPLatencyEntry (0.00s)
    --- PASS: TestLatencyStore_getNodeIPLatencyEntry/10.244.2.1 (0.00s)
    --- PASS: TestLatencyStore_getNodeIPLatencyEntry/10.244.2.2 (0.00s)
=== RUN   TestLatencyStore_SetNodeIPLatencyEntry
=== RUN   TestLatencyStore_SetNodeIPLatencyEntry/10.244.2.1
=== RUN   TestLatencyStore_SetNodeIPLatencyEntry/10.244.2.1#01
--- PASS: TestLatencyStore_SetNodeIPLatencyEntry (0.00s)
    --- PASS: TestLatencyStore_SetNodeIPLatencyEntry/10.244.2.1 (0.00s)
    --- PASS: TestLatencyStore_SetNodeIPLatencyEntry/10.244.2.1#01 (0.00s)
=== RUN   TestLatencyStore_DeleteStaleNodeIPs
--- PASS: TestLatencyStore_DeleteStaleNodeIPs (0.00s)
=== RUN   TestLatencyStore_ListNodeIPs
=== RUN   TestLatencyStore_ListNodeIPs/List_Node_IPs
--- PASS: TestLatencyStore_ListNodeIPs (0.00s)
    --- PASS: TestLatencyStore_ListNodeIPs/List_Node_IPs (0.00s)
PASS
ok      antrea.io/antrea/pkg/agent/monitortool  1.068s
```